include/dynd/kernels/arithmetic.hpp

```diff
@@ -7,42 +7,66 @@
 namespace dynd {
 namespace nd {
 
-  template <type_id_t I0>
-  struct plus_kernel : base_kernel<plus_kernel<I0>, 1> {
+  template <type_id_t I0, typename func_type, func_type f>
+  struct unary_kernel : base_kernel<unary_kernel<I0, func_type, f>, 1> {
     typedef typename type_of<I0>::type A0;
```

Comment:
A few changes I would suggest here. We should rename I0, I1, ... to Src0TypeID, Src1TypeID, ... (this was introduced to the other kernels later on after the arithmetic_kernels were written). We also should write template parameters AsFirstLetterCapitalized. @mwiebe and I decided on this at one point and we're trying to be consistent now.